pkg/bindings/images/build.go

```diff
@@ -298,6 +298,12 @@ func Build(ctx context.Context, containerFiles []string, options entities.BuildO
 	if len(options.Platforms) > 0 {
 		params.Del("platform")
 		for _, platformSpec := range options.Platforms {
+			// podman-cli will send empty struct, in such
+			// case don't add platform to param and let the
+			// build backend decide the default platform.
+			if platformSpec.OS == "" && platformSpec.Arch == "" && platformSpec.Variant == "" {
```

Comment:
This of course fails if any of those are non-null, such as Arch or Variant alone. I don't know if those are possible, or, if they are, if the code below assembles them in the desired fashion ("/arch", "//variant"). Assuming those are OK, this LGTM. Thank you!

Reply:
> This of course fails if any of those are non-null, such as Arch or Variant alone.

In this case it is valid since users are free to choose only `arch` or `/arch`after https://github.com/containers/buildah/pull/4757 and backend will perform `Trim` in appropriate manner ( we have a test case for this on buildah end ). Choosing only variant is invalid and backend will detect and report it as an invalid value after https://github.com/containers/buildah/pull/4757. I hope this answers your concern, let me know if i missed anything :)

Reply:
It would be nice if we would have a unit tested function somewhere that Podman and Buildah can share but I am cool to merge as is.